skimage/filters/ridges.py

```diff
@@ -0,0 +1,651 @@
+                will be removed in version 0.16.""")
+        gamma = beta2
+
+    # Check (sigma) scales
+    sigmas = np.arange(scale_range[0], scale_range[1], scale_step)
+    if np.any(np.asarray(sigmas) < 0.0):
+        raise ValueError("Sigma values less than zero are not valid")
+
+    # Rescale filter parameters
+    alpha = 2 * alpha ** 2
+    beta = 2 * beta ** 2
+    gamma = 2 * gamma ** 2
+
+    # Get image dimensions
+    ndim = image.ndim
+    shape = image.shape
+
+    # Invert image to detect dark ridges on light background
+    if black_ridges:
+        image = invert(image)
+
+    # Generate empty (n+1)D arrays for storing auxiliary images filtered
+    # at different (sigma) scales
+    filtered_array = np.zeros(sigmas.shape + image.shape)
+    lambdas_array = np.zeros(sigmas.shape + image.shape)
+
+    # Filtering for all (sigma) scales
+    for i, sigma in enumerate(sigmas):
+
+        # Calculate (abs sorted) eigenvalues
+        eigenvalues = compute_hessian_eigenvalues(image, sigma, sorting='abs')
+
+        if ndim == 2:
+
+            # Get Hessian eigenvalues
+            (lambda1, lambda2) = eigenvalues
+
+            # Compute sensitivity to deviation from a blob-like structure
+            r_b = _divide_nonzero(lambda1, lambda2) ** 2
+
+            # Compute sensitivity to areas of high variance/texture/structure
+            r_g = lambda1 ** 2 + lambda2 ** 2
+
+            # Compute output image for given (sigma) scale
+            filtered = (np.exp(-r_b / beta)
+                        * (np.ones(shape) - np.exp(-r_g / gamma)))
+
+            # Store results in (2+1)D matrices
+            filtered_array[i] = filtered
+            lambdas_array[i] = lambda2
+
+        elif ndim == 3:
+
+            # Get Hessian eigenvalues
+            (lambda1, lambda2, lambda3) = eigenvalues
+
+            # Compute sensitivity to deviation from a plate-like structure
+            r_a = _divide_nonzero(lambda2, lambda3) ** 2
+
+            # Compute sensitivity to deviation from a blob-like structure
+            r_b = _divide_nonzero(lambda1,
+                                  np.sqrt(np.abs(lambda2 * lambda3))) ** 2
+
+            # Compute sensitivity to areas of high variance/texture/structure
+            r_g = lambda1 ** 2 + lambda2 ** 2 + lambda3 ** 2
+
+            # Compute output image for given (sigma) scale
+            filtered = ((np.ones(shape) - np.exp(-r_a / alpha))
+                        * np.exp(-r_b / beta)
+                        * (np.ones(shape) - np.exp(-r_g / gamma)))
+
+            # Store results in (n+1)D matrices
+            filtered_array[i] = filtered
+            lambdas_array[i] = np.max([lambda2, lambda3], axis=0)
+
+        else:
+
+            # Check image dimensions
+            assert_nD(image, [2, 3])
+
+    # Remove background
+    filtered_array[lambdas_array > 0] = 0
+
+    # Return for every pixel the value of the (sigma) scale with the maximum
```

Comment:
so it's not the value of the scale but the value _at _ the scale (or _for_ the scale? Not sure about the English.)

Reply:
> @DavidBreuer moving the hessian matrix calculations to `_shared` would be a good option IMO. What do you think @jni? We could possibly postpone this refactoring and merge this PR first, but do you think you could to the refactoring yourself later? The most relevant persons would be you and @jni, and @jni has a lot on his plate.

Hi @emmanuelle and @jni , I could definitely try this refactoring in a different, later PR. It feels cleaner to move the Hessian stuff from feature/filters to _shared, if @jni agrees. :)

PS: Right, sorry I'll use mv instead of copy/pasting in the future!

Reply:
By scale you don't mean sigma, right?This is only a hand-wavy explanation: If the lambdas = eigenvalues (lambda_array is a combination of lambdas, depending on 2D/3D) are small, 

http://www.tecn.upf.es/~afrangi/articles/miccai1998.pdf

Reply:
@DavidBreuer I think @emmanuelle's concern here is the language in the comment. I agree with her that it seems to say that we are returning the sigma at which the maximum is achieved. We are not returning that sigma, we are returning the maximum *over all sigmas*.

Reply:
Oh, right, sorry my lack of understanding. I adjusted the explanation which, I agree, was wrong or at least misleading before. Thanks.